src/core/cgroup-bpf-ctx.c

```diff
@@ -23,6 +25,50 @@ static bool program_exists(CGroupContext *c, enum bpf_attach_type attach_type, c
         return false;
 }
 
+int pinned_bpf_program_from_string(const char *str, enum bpf_attach_type *attach_type, char **bpffs_path) {
+        _cleanup_strv_free_ char **parts = NULL;
+        enum bpf_attach_type atype;
+
+        assert(str);
+        assert(bpffs_path);
+        assert(attach_type);
+
+        parts = strv_split(str, ":");
+        if (!parts)
+                return log_oom();
+
+        if (strv_length(parts) < 2)
+                return -EINVAL;
+
+        atype = bpf_cgroup_attach_type_from_string(parts[0]);
+        if ((int) atype < 0)
+                return -EINVAL;
+
+        *attach_type = atype;
+        *bpffs_path = strdup(parts[1]);
```

Comment:
if strdup fails,, the return variable attach_type will be set anyway. Better to strdup first and return log_oom if it fails instead?